xarray/tests/test_merge.py

```diff
@@ -235,6 +235,264 @@ def test_merge_attrs_drop_conflicts(self):
         expected = xr.Dataset(attrs={"a": 0, "d": 0, "e": 0})
         assert_identical(actual, expected)
 
+    def test_merge_attrs_drop_conflicts_non_bool_eq(self):
+        """Test drop_conflicts behavior when __eq__ returns non-bool values."""
+        import warnings
+
+        import numpy as np
```

Comment:
Move these local imports to the top?

Actually, I think we already import numpy, so could perhaps just remove.